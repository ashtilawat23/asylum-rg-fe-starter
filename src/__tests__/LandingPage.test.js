import React from 'react';
import { render } from '@testing-library/react';
import { LandingPage } from '../components/pages/Landing';

test('LandingPage Component renders without errors', () => {
  render(<LandingPage />);
});

test('LandingPage Component matches snapshot', () => {
  const { asFragment } = render(<LandingPage />);
  expect(asFragment()).toMatchSnapshot();
});
