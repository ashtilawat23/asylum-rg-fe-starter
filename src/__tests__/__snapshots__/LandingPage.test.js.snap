// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="header-text-container"
      >
        <h1>
          Asylum Office Grant Rate Tracker
        </h1>
        <h3>
          The Asylum Office Grant Rate Tracker provides asylum seekers, researchers, policymakers, and the public an interactive tool to explore USCIS data on Asylum Office decisions
        </h3>
      </div>
    </div>
    <div
      class="graphs-section"
    >
      <div
        class="graph-container"
      >
        <img
          alt="Bar Graph"
          class="graph"
          src="bar-graph-no-text.png"
        />
        <p>
          Search Grant Rates By Office
        </p>
      </div>
      <div
        class="graph-container"
      >
        <img
          alt="Pie Chart"
          class="graph"
          src="pie-chart-no-text.png"
        />
        <p>
          Search Grant Rates By Nationality
        </p>
      </div>
      <div
        class="graph-container"
      >
        <img
          alt="Line Graph"
          class="graph"
          src="line-graph-no-text.png"
        />
        <p>
          Search Grant Rates Over Time
        </p>
      </div>
    </div>
    <div
      class="view-more-data-btn-container"
    >
      <button
        class="ant-btn ant-btn-default"
        style="background-color: rgb(64, 76, 74); color: rgb(255, 255, 255);"
        type="button"
      >
        <span>
          View the Data
        </span>
      </button>
      <a
        download="COW2021001887-I589Data.csv"
        href="/COW2021001887-I589Data.csv"
      >
        <button
          class="ant-btn ant-btn-default"
          style="background-color: rgb(64, 76, 74); color: rgb(255, 255, 255);"
          type="button"
        >
          <span>
            Download the Data
          </span>
        </button>
      </a>
    </div>
    <div
      class="middle-section"
    >
      <div
        class="hrf-img-container"
      >
        <img
          alt="Human Rights First"
          class="hrf-img"
          src="paper-stack.jpg"
        />
      </div>
      <div
        class="middle-section-text-container"
      >
        <h3>
          Human Rights First has created a search tool to give you a user-friendly way to explore a data set of asylum decisions between FY 2016 and May 2021 by the USCIS Asylum Office, which we received through a Freedom of Information Act request. You can search for information on asylum grant rates by year, nationality, and asylum office, visualize the data with charts and heat maps, and download the data set
        </h3>
      </div>
    </div>
    <div
      class="bottom-section"
    >
      <h3>
        Systemic Disparity Insights
      </h3>
      <div
        class="stats-container"
      >
        <div
          class="stat"
        >
          <h3>
            36%
          </h3>
          <p>
            By the end of the Trump administration, the average asylum office grant rate had fallen 36 percent from an average of 44 percent in fiscal year 2016 to 28 percent in fiscal year 2020.
          </p>
        </div>
        <div
          class="stat"
        >
          <h3>
            5%
          </h3>
          <p>
            The New York asylum office grant rate dropped to 5 percent in fiscal year 2020.
          </p>
        </div>
        <div
          class="stat"
        >
          <h3>
            6x Lower
          </h3>
          <p>
            Between fiscal year 2017 and 2020, the New York asylum office's average grant rate was six times lower than the San Francisco asylum office.
          </p>
        </div>
      </div>
      <div
        class="read-more-btn-container"
      >
        <button
          class="ant-btn ant-btn-default"
          style="background-color: rgb(64, 76, 74); color: rgb(255, 255, 255);"
          type="button"
        >
          <a
            href="https://humanrightsfirst.org/library/uscis-records-reveal-systemic-disparities-in-asylum-decisions/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Read More
          </a>
        </button>
      </div>
    </div>
    <p
      class="back-to-top"
    >
      Back To Top ^
    </p>
  </div>
</DocumentFragment>
`;
